feat: Adding Semantic Text Splitter Component (Text Splitters)

joaoguilhermeS opened the pull request:
**Feat: Semantic Text Splitter with Advanced Threshold Controls**

**Overview**
Introduces a new Semantic Text Splitter component that provides flexible text chunking with statistical threshold controls and regex support.

**Features:**
- Multiple threshold control methods:
  - Percentile-based splitting
  - Standard deviation thresholds
  - Interquartile range
- Configurable chunk size and count
- Optional regex-based splitting

edwinjosechittilappilly reviewed (changes requested):
@joaoguilhermeS  is langchain_experimental a part of the pyproject of langflow? if not @ogabrielluiz can we add it to the pyproject?
Can you confirm if its already added by uv to the pyproject file?

Also I suggest we add the beta as True in component since its a part of langchain_experimental

joaoguilhermeS commented:
Thanks for your review @edwinjosechittilappilly I have added the langchain_experimental to the pyproject and everything seems to be working. I also added the beta flag on the Semantic Text Splitter component.

edwinjosechittilappilly commented:
@joaoguilhermeS Great job.

https://github.com/langflow-ai/langflow/blob/jg/feat-semantic-text-splitter/src/backend/base/pyproject.toml
I already have langchain_experimental.

Can you test adding a different version of langchain_experimental in both?

I suggest updating the **base pyproject** so we won’t need to add it in other pyprojects outside of the base.

ogabrielluiz commented on pyproject.toml:
```diff
@@ -109,6 +109,7 @@ dependencies = [
     "duckduckgo-search>=6.3.0",
     "langchain-elasticsearch>=0.2.0",
     "opensearch-py>=2.7.1",
+    "langchain-experimental>=0.3.2"
```
We shouldn't need to add this as we have it as dependency in Langflow-base

joaoguilhermeS replied:
I see, you are right. I just removed it and kept it in the base. Thanks for the insight.

joaoguilhermeS commented:
Hey @edwinjosechittilappilly, I tested the component with the current langchain-experimental dependency version and it is working just fine, so I think there is no need to update so we do not break other components. Thanks for the review.

edwinjosechittilappilly reviewed (approved):
LGTM.
@Cristhianzl  we might need to update the uvlock accordingly later if required.

edwinjosechittilappilly reviewed (changes requested):
@joaoguilhermeS  if there is no change in the dependencies then I would suggest the uvlock need not be updated.

Apart from that its all good.

edwinjosechittilappilly reviewed (approved):
LGTM

edwinjosechittilappilly commented:
@ogabrielluiz I believe you might need to approve for this to get merged.(Since changes requested). Can you take a look into it, Its been blocked from merging.
